### PR TITLE
Migrate api module's build file to kotlin dsl

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description
Use kotlin dsl in the build file of api module. Same as previous one, two commits one to rename and second with the actual migration(the second commit is the one to review).
Note: Should be squashed(went through the git history and there's nothing interesting, I'll manually go back and fix any issues if needed).

## How Has This Been Tested?

Ran the build/test tasks + other tasks defined in the build file and verified their output.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: api/build.gradle.kts [hunk 1]
```diff
@@ -1,139 +1,154 @@
...
+        minSdk = 16
         //noinspection OldTargetApi
-        targetSdk 32
-        buildConfigField "String", "READ_WRITE_PERMISSION", '"com.ichi2.anki.permission.READ_WRITE_DATABASE"'
-        buildConfigField "String", "AUTHORITY", '"com.ichi2.anki.flashcards"'
+        targetSdk = 32
+        buildConfigField(
+            "String",
+            "READ_WRITE_PERMISSION",
+            "\"com.ichi2.anki.permission.READ_WRITE_DATABASE\""
```
Comment (david-allison) [inline api/build.gradle.kts:29]: This feels worse, is there a better way to do this?

File: api/build.gradle.kts [hunk 2]
```diff
@@ -1,139 +1,154 @@
...
+val dokkaJavadocJar = tasks.register("dokkaJavadocJar", Jar::class) {
+    from(tasks.named<DokkaTask>("dokkaJavadoc"))
     archiveClassifier.set("javadoc")
     doLast {
-        println("API javadocs output directory: ${dokkaJavadoc.outputDirectory.get()}")
+        println("API javadocs output directory: ${tasks.named<DokkaTask>("dokkaJavadoc").get().outputDirectory.get()}")
         println("API javadocs jar output directory: ${destinationDirectory.get()}")
     }
 }
+tasks.named("dokkaJavadocJar").dependsOn(tasks.named<DokkaTask>("dokkaJavadoc"))
```
Comment (david-allison) [inline api/build.gradle.kts:87]: can we extract `tasks.named<DokkaTask>("dokkaJavadoc")`

**Review by david-allison — APPROVED** [review]
Two optional nitpicks

File: api/build.gradle.kts (hunk 1 above)
Comment (lukstbit) [reply api/build.gradle.kts:29]: We could use kotlin multiline strings but somehow:

`""""com.ichi2.anki.permission.READ_WRITE_DATABASE""""`

doesn't feel better.

**Review by BrayanDSO — APPROVED** [review]

**Pull request merged**